From 0f45942f6277754f20ca83daf397d3afa7409e57 Mon Sep 17 00:00:00 2001
From: "chengshun.wang" <chengshun.wang@amlogic.com>
Date: Thu, 13 Feb 2014 14:39:12 +0800
Subject: [PATCH 3264/5965] PD #85968: rt5616 record had noise when all audio
 codec use one machine

PD# 85968: change the way to compare codec name
---
 sound/soc/aml/m8/aml_m8.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/sound/soc/aml/m8/aml_m8.c b/sound/soc/aml/m8/aml_m8.c
index 3ec1a27f8193..2b964fda50fe 100755
--- a/sound/soc/aml/m8/aml_m8.c
+++ b/sound/soc/aml/m8/aml_m8.c
@@ -242,8 +242,13 @@ static int aml_asoc_hw_params(struct snd_pcm_substream *substream,
     }
 
     /* set cpu DAI configuration */
-    ret = snd_soc_dai_set_fmt(cpu_dai, SND_SOC_DAIFMT_I2S |
-        SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_CBM_CFM);
+    if(!strncmp(codec_info.name_bus,"rt5616",6)){
+        ret = snd_soc_dai_set_fmt(cpu_dai, SND_SOC_DAIFMT_I2S |
+            SND_SOC_DAIFMT_IB_NF | SND_SOC_DAIFMT_CBM_CFM);
+    }else{
+        ret = snd_soc_dai_set_fmt(cpu_dai, SND_SOC_DAIFMT_I2S |
+            SND_SOC_DAIFMT_NB_NF | SND_SOC_DAIFMT_CBM_CFM);
+    }
     if (ret < 0) {
         printk(KERN_ERR "%s: set cpu dai fmt failed!\n", __func__);
         return ret;
-- 
2.19.0

